lib/pos_blockchain.go

```diff
@@ -453,6 +498,24 @@ func (bc *Blockchain) validatePreviouslyIndexedBlockPoS(blockHash *BlockHash) (*
 		return nil, errors.Wrapf(err, "validatePreviouslyIndexedBlockPoS: Problem fetching block from DB")
 	}
 
+	// If the block isn't validated or validate failed, we need to run the anti-spam checks on it.
+	failedSpamPreventionCheck, err := bc.validateLeaderAndQC(block)
+	if err != nil {
+		// If we hit an error, that means there was an intermittent issue when trying to
+		// validate the QC or the leader. This is a really bad error to hit...
+		return nil, errors.Wrap(err, "validatePreviouslyIndexedBlockPoS: Problem validating leader and QC")
+	}
+	if failedSpamPreventionCheck {
```

Comment:
Followup to the above passedSpamPreventionCheck comment assuming you accept my suggested chain
```suggestion
	if !passedSpamPreventionCheck {
```